tokio/tests/task_local_set.rs

```diff
@@ -573,6 +573,25 @@ async fn spawn_wakes_localset() {
     }
 }
 
+#[tokio::test]
+async fn sleep_with_local_enter_guard() {
```

Comment:
This isn't a big deal, but would you mind maybe adding a comment describing what this test reproduces (and maybe referecning issue https://github.com/tokio-rs/tokio/issues/5020)?

Reply:
Okay! I've added.